scripts/imgtool/main.py

```diff
@@ -105,6 +107,27 @@ def validate_header_size(ctx, param, value):
             "Minimum value for -H/--header-size is {}".format(min_hdr_size))
     return value
 
+def get_dependencies(ctx, param, value):
+    if value is None:
+        return None
+    else:
+        versions = []
+        images = re.findall("\((\d+)", value)
+        raw_versions = re.findall(",\s*([0-9.+]+)\)", value)
+        if len(images) != len(raw_versions):
+            raise click.BadParameter(
+                "There's a mismatch between the number of \
+                dependency images and versions in: {}".format(value))
+        for raw_version in raw_versions:
+            try:
+                versions.append(decode_version(raw_version))
+            except ValueError as e:
+                raise click.BadParameter("{}".format(e))
+        if len(images) != 0:
```

Comment:
You should throw `click.BadParameter` if `len(images) == 0` before the other tests above; so this test ^ can be removed and you wouldn't die trying to return a non existing `dependencies` variable below.

Reply:
Btw, would be interesting to document how the dependencies are to be specified in the help page...